node/src/node_configurator/configurator.rs

```diff
@@ -2583,7 +2583,7 @@ mod tests {
                 clandestine_port: 1234,
                 chain_name: "ropsten".to_string(),
                 gas_price: 2345,
-                max_block_count_opt: Some(100000),
+                max_block_count_opt: None,
```

Comment:
@dnwiebe Had feedback I hadn't covered the usage of `None` so now there's one for `Some(100_000u64)` and this one for `None.

Reply:
Yes, it kind of makes sense but I also think this solution is sub-optimal. Somebody who wouldn't know about these intentions could easily change the value on a sudden occasion and this fragile warranty will break up. I know this isn't really your card, this is a long time system issue, but now I think we should've proceed by creating the single test claiming to test all optional parameters when they evaluate to `None`. It would've laid down a path for any time when we need to deal with this in the future because hopefully the test would make the coder have to modify it with every new parameter and he should first go look at the title of the test. If written clearly, it should help the coder understand that he's supposed to set his newly added parameter with `None` in this test. (While he should use Some in the others). This could bring us more assurance that the None won't lose because the purpose of the test will stand above as a sign of undesirability in the removal of it.

I don't know I feel like I would want too much at the moment if I was insisting on this. So it's in your hands. I'm gonna give you an approval now (and another if you decided to write such a test covering all parameters with optional values).

Reply:
I kind of thought thats what the test below was doing but you recommended deleting it. I can restore it of course but I need some guidance on how to improve it where you thought it was lacking. If the test name `configuration_handles_retrieving_max_block_count_none` is too specific I could change it to `configuration_handles_retrieving_all_possible_none_values' or something.